pkg/controller/sparkapplication/controller.go

```diff
@@ -401,8 +406,10 @@ func (c *Controller) getAndUpdateExecutorState(app *v1beta2.SparkApplication) er
 					c.recordExecutorEvent(app, newState, pod.Name)
 				}
 			}
-			executorStateMap[pod.Name] = newState
-
+			//if the executor number is higher than the `executorsProcessingLimit` we want to stop persisting executors
```

Comment:
```suggestion
			// If the executor number is higher than the `executorsProcessingLimit` we want to stop persisting executors
```

Reply:
Also:
This comment does not actually match with the code below, you don't stop persisting executor if we cross the threshold, we only stop the tracking of executors above the threshold

Reply:
Did you consider that we could also disable this at the beginning of the function with `disableExecutorReporting` by checking that `len(app.Status.ExecutorState) < executorsProcessingLimit`

Reply:
We didn't want to disable the executor reporting at all and continue sending k8s events, for example, but we wanted to stop storing the executors in the spark CR in case of a significant amount of executors, this behavior is very unlikely, for example, unhandled executor storm.

But probably your advice is better. Since the situation is unlikely we can just disable the executor reporting at all